AndroidNativeLoginTemplate/app/src/main/java/com/salesforce/androidnativelogintemplate/NativeLogin.kt

```diff
@@ -161,167 +199,516 @@ class NativeLogin : ComponentActivity() {
+        ) {
+            OutlinedTextField(
+                value = username,
+                onValueChange = { username = it },
+                label = { Text("Username") },
+            )
 
-        Row(content = {
-            Button(
-                onClick = {
-                    loading = true
-                    scope.launch { loading = login(username, password) }
-                },
-                modifier = Modifier.width(150.dp),
-            ) {
-                Text(text = "Login")
+            OutlinedTextField(
+                value = password,
+                onValueChange = { password = it },
+                label = { Text("Password") },
+                visualTransformation = PasswordVisualTransformation(),
+                keyboardOptions = KeyboardOptions(keyboardType = Password),
+            )
+
+            Spacer(modifier = Modifier.height(16.dp))
+
+            Row(content = {
+                Button(
+                    onClick = {
+                        loading = true
+                        scope.launch { loading = login(username, password) }
+                    },
+                    modifier = Modifier.width(150.dp),
+                ) {
+                    Text(text = "Login")
+                }
             }
+            )
         }
-        )
     }
-}
 
-@Composable
-fun LoginTheme(composable: @Composable () -> Unit) {
-    val dynamicColors = Build.VERSION.SDK_INT >= Build.VERSION_CODES.S
-    val isDarkTheme = isSystemInDarkTheme()
-    val colorScheme = when {
-        dynamicColors && isDarkTheme -> {
-            dynamicDarkColorScheme(LocalContext.current)
-        }
-        dynamicColors && !isDarkTheme -> {
-            dynamicLightColorScheme(LocalContext.current)
+    @Composable
+    fun LoginTheme(composable: @Composable () -> Unit) {
+        val dynamicColors = SDK_INT >= S
+        val isDarkTheme = isSystemInDarkTheme()
+        val colorScheme = when {
+            dynamicColors && isDarkTheme -> {
+                dynamicDarkColorScheme(LocalContext.current)
+            }
+
+            dynamicColors && !isDarkTheme -> {
+                dynamicLightColorScheme(LocalContext.current)
+            }
+
+            !dynamicColors && isDarkTheme -> {
+                darkColorScheme()
+            }
+
+            else -> lightColorScheme()
         }
-        !dynamicColors && isDarkTheme -> {
-            darkColorScheme()
+
+        MaterialTheme(colorScheme = colorScheme, content = composable)
+    }
+
+    @Preview
+    @Composable
+    fun LoginPreview() {
+        Column {
+            LoginView(
+                login = { _, _ -> run { return@LoginView true } },
+                shouldShowBack = true,
+                backAction = {}
+            )
         }
-        else -> lightColorScheme()
     }
 
-    MaterialTheme(colorScheme = colorScheme, content = composable)
-}
+    @Preview
+    @Composable
+    fun LoginViewInitializePasswordLessLoginViaOtpPreview() {
```

Comment:
I added two new previews so one can easily see the request and submit OTP layouts.